### PR TITLE
min_witness_weight param for fundpsbt/utxopsbt

### PR DESCRIPTION
**Pull request by niftynei**
Add a `min_witness_weight` param to our 'fund-/utxo-psbt' methods. This is to support the dual funding spec, which requires a floor of witness fees to be paid at 110 weight.

In theory, if we ever get more complicated UTXOs, we'll want to update the witness weight calculation to encompass the correct weight for these, also.

### CODE REVIEW

**Comment by rustyrussell** [comment]
Since the difference is small, I wonder if we should just set this to 110 always, and note that this is a requirement to comply with the df spec?

File: wallet/reservation.c [hunk 1]
```diff
@@ -427,7 +429,9 @@ static struct command_result *json_fundpsbt(struct command *cmd,
 						    "impossible UTXO value");
 
 			/* But also adds weight */
-			*weight += utxo_spend_weight(utxo);
+			fprintf(stderr, "min_witness_weight is %u\n",
+					*min_witness_weight);
```
Comment (cdecker) [inline wallet/reservation.c:433]: Debugging statement.

**Comment by niftynei** [comment]
> Since the difference is small, I wonder if we should just set this to 110 always, and note that this is a requirement to comply with the df spec?

We could, but at the cost (haha) of precision in feerate on non-dual-funded transactions.

**Comment by niftynei** [comment]
Merged elsewhere.

**Pull request closed without merging**